src/chunk_constraint.c

```diff
@@ -1290,6 +1290,8 @@ check_chunk_constraint_violated(Oid chunk_relid, const Dimension *dim, const Dim
 	TupleTableSlot *slot;
 	TableScanDesc scandesc;
 	bool isnull;
+	int attno = get_attnum(chunk_relid, NameStr(dim->fd.column_name));
+	Assert(attno != InvalidAttrNumber);
```

Comment:
Since this is from the metadata it can actually be corrupt for unknown reasons, and `Ensure` is probably a better option here. We want to have a runtime check if something is broken in a release build.